experimental/storage/src/Storage.ts

```diff
@@ -0,0 +1,9 @@
+import { UploadType } from "./upload/Upload";
+import { Upload } from "./upload/upload";
+
+export class Storage implements Storage {
+  constructor() {}
+  upload(params: UploadType.Options): Upload {
+    return new Upload(params);
+  }
+}
```

Comment:
Any reason why Storage should be a class? Is there any status shared between uploads? If it only acts as a namespace, we can use a constant or namespace. Excessive `new` can be annoying to some JS devs. I don't see the possiblity that we need construtor parameter on this level.

Reply:
Yeah we totally can. Would you like to do something like this instead? 
```
new Storage.Upload({...})
```